Kitodo/src/main/java/org/kitodo/production/model/Subfolder.java

```diff
@@ -303,6 +306,31 @@ public Optional<URI> getURIIfExists(String canonical) {
         return new File(uri.getPath()).isFile() ? Optional.of(uri) : Optional.empty();
     }
 
+    /**
+     * Checks if the folder determined by determineDirectoryAndFileNamePattern() is empty.
+     * Retrieve the directory URI and lists its contents.
+     * It stops checking as soon as the first file is found.
+     *
+     * @return true if the folder is empty, false if it contains at least one file
+     */
+    public boolean isFolderEmpty() {
+        Pair<URI, Pattern> query = determineDirectoryAndFileNamePattern();
+        File dir = fileService.getFile(query.getLeft());
+        if (!dir.exists() || !dir.isDirectory()) {
+            return false;  // Ensures we only check valid directories
+        }
+        try (Stream<Path> entries = Files.list(dir.toPath())) {
+            // Apply regex filtering to check only for files matching the pattern
+            return !entries
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .anyMatch(name -> query.getRight().matcher(name).matches()); // Stop after first match
```

Comment:
IntelliJ IDEA suggest here instead of `!entries.anyMatch()` to use `entries.noneMatch()` to use.

Reply:
I got the same recommendation. I was just wondering wether entries.noneMatch() is slower here. But maybe we should trust the IDE more.

Reply:
I noticed no differences on my local development system. Maybe you can time it on your system with an S3 in background? Maybe the first solution was faster but maybe there is no difference.

Reply:
There seem to be no real performance differences
https://stackoverflow.com/a/57779846